### PR TITLE
[release] Initiate 7.7 branch

### PR DESCRIPTION
**Pull request by mgreau**
This commit initiates the `7.7` branch which will be dedicated to the `7.7.x` releases for the Helm Charts. It will allow to:
- test this branch with the daily Stack Docker images 7.7.0-SNAPSHOT via dedicated Jenkins jobs
- test the staging 7.7.x Docker images before a release

The files related to the 6.x versions are removed from this branch.
The `CHANGELOG` file is also removed, the intent is to have a unique file for dealing with the changelogs on `master`.

The doc still needs to be updated to introduce this new workflow.

### CODE REVIEW

File: elasticsearch/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ This helm chart is a lightweight way to configure and run our official [Elastics
 
 ## Notice
 
-[7.6.1](https://github.com/elastic/helm-charts/releases/tag/7.6.1) release is introducing a change for Elasticsearch users upgrading from a previous chart version.
+[7.7.0-SNAPSHOT](https://github.com/elastic/helm-charts/releases/tag/7.7.0-SNAPSHOT) release is introducing a change for Elasticsearch users upgrading from a previous chart version.
```
Comment (jmlrt) [inline elasticsearch/README.md]: version shouldn't be bump in this warning

File: .ci/jobs.t/elastic+helm-charts+{branch}+staging+cluster-cleanup.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 - job:
-    name: elastic+helm-charts+staging+cluster-cleanup
+    name: elastic+helm-charts+%BRANCH%+staging+cluster-cleanup
```
Comment (jmlrt) [inline .ci/jobs.t/elastic+helm-charts+{branch}+staging+cluster-cleanup.yml]: Staging jobs where created by #532 to tests staging Docker image. IMHO these jobs should stay only on master branch.

File: .ci/jobs/elastic+helm-charts+pull-request+cluster-cleanup.yml [hunk 3]
```diff
@@ -1,35 +0,0 @@
----
```
Comment (jmlrt) [inline .ci/jobs/elastic+helm-charts+pull-request+cluster-cleanup.yml]: Will pull-request jobs on master branch be able to test also pull requests for other branches or should we have some pull-request job for each release branch?

File: .ci/jobs.t/elastic+helm-charts+{branch}+staging+cluster-cleanup.yml (hunk 2 above)
Comment (mgreau) [reply .ci/jobs.t/elastic+helm-charts+{branch}+staging+cluster-cleanup.yml:3]: With this config we should automatically have staging jobs generated per git branch. Those "template" files will be present on all git branches (without needed to update them)

File: .ci/jobs/elastic+helm-charts+pull-request+cluster-cleanup.yml (hunk 3 above)
Comment (mgreau) [reply .ci/jobs/elastic+helm-charts+pull-request+cluster-cleanup.yml:1]: I'm not sure about the behavior with PRs against this branch with this new config. For now, I have removed those `jjb` files and I would like to first merge this PR and make sure all the `7.7` jobs are OK (CI/staging) before dealing with the PRs jobs

**Comment by mgreau** [comment]
jenkins test this please

**Pull request merged**